test/e2e/specs/wp-calypso-gutenberg-checkout-spec.js

```diff
@@ -0,0 +1,260 @@
+				'Coupon not removed properly'
+			);
+		} );
+
+		step( 'Can Save Order And Continue', async function () {
+			return await driverHelper.clickWhenClickable(
+				driver,
+				By.css(
+					'.composite-checkout .wp-checkout__review-order-step .checkout-button.is-status-primary'
+				)
+			);
+		} );
+	} );
+
+	describe( 'Can make payment', function () {
+		const testCreditCardDetails = dataHelper.getTestCreditCardDetails();
+
+		step( 'Can fill out billing information', async function () {
+			const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+			return await securePaymentComponent.completeTaxDetailsInContactSection(
+				testCreditCardDetails
+			);
+		} );
+
+		step( 'Can select and fill out credit card payment method', async function () {
+			const existingCardIsPresent = await driverHelper.isElementPresent(
+				driver,
+				By.css( '[id*="existingCard-"]:checked' )
+			);
+
+			if ( ! existingCardIsPresent ) {
+				const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+				return await securePaymentComponent.enterTestCreditCardDetails( testCreditCardDetails );
+			}
+
+			return true;
+		} );
+
+		step( 'Can process the payment', async function () {
+			const securePaymentComponent = await SecurePaymentComponent.Expect( driver );
+			await securePaymentComponent.submitPaymentDetails();
+			await securePaymentComponent.waitForCreditCardPaymentProcessing();
+			const errorNoticeIsNotPresent = await driverHelper.elementIsNotPresent(
+				driver,
+				By.css( '.notice.is-error' )
+			);
+			assert.strictEqual(
+				errorNoticeIsNotPresent,
+				true,
+				'The checkout produced a notice error when processing this payment'
+			);
+			return await securePaymentComponent.waitForPageToDisappear();
+		} );
+
+		step( 'Can decline upgrade offer', async function () {
+			const upsellPage = await UpsellPage.Expect( driver );
+			return await upsellPage.declineOffer();
+		} );
+
+		step( 'Can return to editor', async function () {
+			const gEditorComponent = await GutenbergEditorComponent.Expect( driver );
+			await gEditorComponent.initEditor();
+			await driver.switchTo().defaultContent();
+			const destinationUrl = await driver.executeScript( 'return window.location.href' );
+			assert.strictEqual(
+				editorUrl,
+				destinationUrl,
+				'Did not return to editor after using editor checkout'
+			);
+		} );
+	} );
+
+	describe( 'Can delete the premium plan', async function () {
+		step( 'Can log in', async function () {
+			const loginFlow = new LoginFlow( driver, 'gutenbergSimpleSiteFreePlanUser' );
+			return await loginFlow.login();
+		} );
+
+		step( 'We can set the sandbox cookie for payments', async function () {
+			const wPHomePage = await WPHomePage.Visit( driver );
+			await wPHomePage.checkURL( locale );
+			await wPHomePage.setSandboxModeForPayments( sandboxCookieValue );
+			await wPHomePage.setCurrencyForPayments( currencyValue );
+			return await driver.navigate().back();
+		} );
```

Comment:
Probably we don't want to log in again but just navigate from Editor back to Calypso so the `DeletePlanFlow` can do the rest. Not a blocker for this PR.